janus/retrieval/retriever.py

```diff
@@ -8,7 +8,10 @@ def __init__(self, collection_name : str, client : chromadb.HttpClient) -> None:
         self.collection = self.client.get_collection(name=self.collection_name) 
 
     def retrieve(self, query : str, n_results : int = 3) -> list:
-        return self.collection.query(query_texts=[query], 
+        result = self.collection.query(query_texts=[query], 
                                      n_results=n_results, 
                                      include=['documents', 'metadatas'])
+        result_filenames = [entry['filename'] for entry in result['metadatas'][0]]
+        names_and_documents = tuple(zip(result_filenames, result['documents'][0]))
+        return names_and_documents
```

Comment:
This makes sense to me. I think this changes the function's return type from a dictionary to a tuple. Is that ok? I actually got a little mixed up, because it seems like the caller of this function expects the function to return a string? https://github.com/spear-ai/janus-llm/blob/public/janus/translate.py#L324

Reply:
You're correct in your line of thought. My tests have passed fine, and I suspect that langchain's prompting method .format_prompt that get called down the line will cast whatever is passed to it as a string, but I can also just do the casting here for consistency's sake. TLDR things are functional right now, but if you prefer the casting to happen within retreive.py I can make that fix.

Reply:
That's fine by me! If this were typed it could return a Union of string, tuple, whatever.